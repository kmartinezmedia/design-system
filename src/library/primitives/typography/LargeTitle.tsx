import React, { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';
import {
  fontFamily,
  fontWeight,
  fontSize,
  lineHeight,
} from '@designSystem/theme';

import { Text, TextProps } from './Text';

export const LargeTitle: FunctionComponent<TextProps> = ({
  children,
  spacing = { bottom: 8 },
  ...textProps
}) => (
  <Text style={styles.LargeTitle} spacing={spacing} {...textProps}>
    {children}
  </Text>
);

const styles = StyleSheet.create({
  LargeTitle: {
    fontSize: fontSize.largeTitle,
    fontFamily: fontFamily.regular,
    fontWeight: fontWeight.regular,
    lineHeight: lineHeight.largeTitle,
  },
});
