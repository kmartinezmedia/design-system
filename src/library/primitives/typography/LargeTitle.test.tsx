import React from 'react';
import 'react-native';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import { LargeTitle } from './LargeTitle';

describe('LargeTitle', () => {
  it('default', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<LargeTitle>Hey, lil 🐛</LargeTitle>);
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
