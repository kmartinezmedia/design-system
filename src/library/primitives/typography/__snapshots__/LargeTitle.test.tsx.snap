// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeTitle default 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Graphik-Regular",
        "fontSize": 64,
        "fontWeight": "400",
        "lineHeight": 88,
      },
      Object {
        "color": "#071422",
        "lineHeight": 88,
        "textAlign": undefined,
      },
      Object {
        "paddingBottom": 32,
      },
      Object {},
      Object {},
    ]
  }
>
  Hey, lil 🐛
</Text>
`;
